more info about how to scrape k0s internals (#3915)
k0sproject/k0s: docs/system-monitoring.md

sudo k0s install controller --enable-metrics-scraper
```

> Once you enable it new set of objects will show up on your cluster:

Reviewer: ```suggestion
Once enabled, a new set of objects will appear in the cluster:
```

```shell
❯ ~ kubectl get all -n k0s-system
NAME                                   READY   STATUS    RESTARTS   AGE
pod/k0s-pushgateway-6c5d8c54cf-bh8sb   1/1     Running   0          43h

NAME                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/k0s-pushgateway   ClusterIP   10.100.11.116   <none>        9091/TCP   43h

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/k0s-pushgateway   1/1     1            1           43h

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/k0s-pushgateway-6c5d8c54cf   1         1         1       43h
```

> That's not enough to start scraping these additional metrics.
> 
> For Prometheus based solution you can create ServiceMonitor for it like this:

Reviewer: ```suggestion
That's not enough to start scraping these additional metrics. For Prometheus
Operator](https://prometheus-operator.dev/) based solutions, you can create a
`ServiceMonitor` for it like this:
```

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: k0s
  namespace: k0s-system
spec:
  endpoints:
  - port: http
  selector:
    matchLabels:
      app: k0s-observability
      component: pushgateway
      k0s.k0sproject.io/stack: metrics
```

> Note that it won't clear alerts like "KubeControllerManagerDown" nor "KubeSchedulerDown" as they are based on Prometheus's internal "up" metrics.
> But you can get rid of these alerts by modifying them to detect working component like this:

Reviewer: ```suggestion
Note that it won't clear alerts like "KubeControllerManagerDown" or
"KubeSchedulerDown" as they are based on Prometheus' internal "up" metrics. But
you can get rid of these alerts by modifying them to detect a working component
like this:
```
